cmd/traffic/cmd/manager/internal/cluster/info.go

```diff
@@ -178,15 +180,38 @@ func (oi *info) watchSubnets(ctx context.Context, name, kind string, retriever f
 			case <-ctx.Done():
 				return
 			case <-acc.Changed():
-				if oi.accUpdate(ctx, acc) {
-					oi.PodSubnets = retriever(ctx)
+				if oi.accUpdate(ctx, acc) && oi.updateSubnets(retriever(ctx)) {
 					oi.waiter.Broadcast()
 				}
 			}
 		}
 	}()
 }
 
+func (oi *info) updateSubnets(podCIDRs []*net.IPNet) bool {
+	changed := true
+	if len(podCIDRs) == len(oi.podCIDRMap) {
+		changed = false // assume equal
+
+		// Check if all IPs are found and that their masks are equal
+		for _, cidr := range podCIDRs {
+			if ones, ok := oi.podCIDRMap[iputil.IPKey(cidr.IP)]; ok {
+				newOnes, _ := cidr.Mask.Size()
+				if newOnes == ones {
+					continue
+				}
+			}
+			changed = true
+			break
+		}
+	}
+	if changed {
+		oi.podCIDRMap = makeCIDRMap(podCIDRs)
+		oi.PodSubnets = toRPCSubnets(podCIDRs)
```

Comment:
Having these two different views of the same data seems funny to me.  I'd almost rather just have `.podCIDRMap` and call `toRPCSubnets` in the RPC handler.

Reply:
I was thinking of performance (fast comparison using hashmap) and scalability. The data is kept for all clients, which in a big installation might be hundreds. Why trigger hundreds of conversions of the same map in the RPC handler?